canard.c

```diff
@@ -206,14 +197,15 @@ int16_t canardRequestOrRespond(CanardInstance* ins,
     {
         return -CANARD_ERROR_INVALID_ARGUMENT;
     }
-    if (canardGetLocalNodeID(ins) == 0)
+    if (canardGetLocalNodeID(ins) == CANARD_BROADCAST_NODE_ID) 
     {
+        // Service transfer not supported for anonymous nodes 
         return -CANARD_ERROR_NODE_ID_NOT_SET;
     }
```

Comment:
There is a problem with the original code: we don't seem to be checking `destination_node_id`, `inout_transfer_id`, `priority`, `kind`, and `payload_len` (against `CANARD_MAX_TRANSFER_PAYLOAD_LEN`). This also applies to other API functions, perhaps we should be more stringent? Or it could be implemented in a dedicated PR?

Reply:
I would suggest move it to another PR as this one is mainly focused on updating the header format